langserver/tracer.py

```diff
@@ -0,0 +1,27 @@
+import opentracing
+import lightstep
+
+
+class Tracer:
+
+    # just use class vars; no need for multiple independent tracer instances
+    lightstep_project = None
+    lightstep_token = None
+
+    @staticmethod
+    def setup(lightstep_project, lightstep_token):
+
+        Tracer.lightstep_project = lightstep_project
+        Tracer.lightstep_token = lightstep_token
+
+        # if these variables aren't set, we'll fall back on the default no-op opentracing implementation
+        if lightstep_project and lightstep_token:
+            opentracing.tracer = lightstep.Tracer(component_name="python-langserver", access_token=lightstep_token)
+
+
+    @staticmethod
+    def start_span(opname, parent_span=None):
```

Comment:
so the only reason this exists is a helper which does the right thing if parent_span is `None`? Do we need that helper? Could we not just rely on `opentracing.tracer` and then assume every span is a child_span excluding when we start a request.